test/controller/error.controller.spec.unit.ts

```diff
@@ -1,23 +1,62 @@
 jest.mock("ioredis");
+jest.mock("../../src/middleware/authentication.middleware");
 jest.mock("../../src/middleware/session.middleware");
+jest.mock("../../src/middleware/objection.session.middleware");
 
 import { NextFunction, Request, Response } from "express";
 import request from "supertest";
 import app from "../../src/app";
+import { ErrorMessages } from "../../src/model/error.messages";
+import { authenticationMiddleware } from "../../src/middleware/authentication.middleware";
 import { sessionMiddleware } from "../../src/middleware/session.middleware";
-import { OBJECTIONS_COMPANY_NUMBER } from "../../src/model/page.urls";
+import { OBJECTIONS_COMPANY_NUMBER, STRIKE_OFF_OBJECTIONS } from "../../src/model/page.urls";
 import { COOKIE_NAME } from "../../src/utils/properties";
+import { Session } from "ch-node-session-handler/lib/session/model/Session";
+import { objectionSessionMiddleware } from "../../src/middleware/objection.session.middleware";
+import { OBJECTIONS_SESSION_NAME } from "../../src/constants";
 
 const mockSessionMiddleware = sessionMiddleware as jest.Mock;
-mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
-  return next(new Error("Error"));
-});
+const mockObjectionSessionMiddleware = objectionSessionMiddleware as jest.Mock;
+const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next());
 
+const DODGY_URL = STRIKE_OFF_OBJECTIONS  + "/company-numberhh";
 const ERROR_500 = "Sorry, there is a problem with the service";
 
 describe("error controller", () => {
 
+  it("should render the 404 template if a page is not found", async () => {
+    mockSessionMiddleware.mockReset();
+    mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
```

Comment:
i would use mockImplementationOnce instead of mockImplementation as we are defining them inside this test, we only need the implementation to be applied just for this test. If we don't, the implementation will apply to the tests that run after (unless they explicitly override the implementation)

Reply:
Ok